SMT tasks should not be descheduled under the same conditions as when tasks
are not idleprio_suitable.

-ck

---
 kernel/sched/bfs.c |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-3.18-ck1/kernel/sched/bfs.c
===================================================================
--- linux-3.18-ck1.orig/kernel/sched/bfs.c	2014-12-31 10:00:08.399896557 +1100
+++ linux-3.18-ck1/kernel/sched/bfs.c	2014-12-31 10:00:59.415898134 +1100
@@ -793,6 +793,8 @@ static bool smt_should_schedule(struct t
 		return true;
 	if (rt_task(p))
 		return true;
+	if (!idleprio_suitable(p))
+		return true;
 	best_bias = best_smt_bias(cpu);
 	/* The smt siblings are all idle or running IDLEPRIO */
 	if (best_bias < 1)
